### PR TITLE
feat(connector): (checkout_dot_com) add support for multiple partial captures

### PR DESCRIPTION
**Pull request by hrithikesh026**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Checkout doesn't support multiple partial captures currently. This PR brings support for multiple partial capture in checkout.

This issue needs to be addressed
https://github.com/juspay/hyperswitch/issues/2036

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Manual

Create payment with `capture_method` = `manual_multiple`
Screenshot 2023-08-21 at 10 50 45 PM

Partially capture the payment (status changes to `partially_captured`)
Screenshot 2023-08-21 at 10 51 09 PM

Capture remaining amount and status changes to `succeeded`
Screenshot 2023-08-21 at 10 51 23 PM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/checkout/transformers.rs [hunk 1]
```diff
@@ -581,17 +582,29 @@ impl TryFrom<&types::PaymentsCaptureRouterData> for PaymentCaptureRequest {
         let connector_auth = &item.connector_auth_type;
         let auth_type: CheckoutAuthType = connector_auth.try_into()?;
         let processing_channel_id = auth_type.processing_channel_id;
+        let capture_type = if item.request.multiple_capture_data.is_some() {
```
Comment (ArjunKarthik) [inline crates/router/src/connector/checkout/transformers.rs:585]: Create a util method is_multiple_capture() -> bool for types::PaymentsCaptureRouterData in `crates/router/src/connector/utils.rs`.

File: crates/router/src/connector/checkout/transformers.rs [hunk 2]
```diff
@@ -609,16 +622,21 @@ impl TryFrom<types::PaymentsCaptureResponseRouterData<PaymentCaptureResponse>>
         } else {
             (enums::AttemptStatus::Pending, None)
         };
+
+        let resource_id = if item.data.request.multiple_capture_data.is_some() {
+            item.response.action_id
```
Comment (ArjunKarthik) [inline crates/router/src/connector/checkout/transformers.rs:627]: Let's have discussion on this

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/checkout/transformers.rs [hunk 3]
```diff
@@ -609,16 +622,21 @@ impl TryFrom<types::PaymentsCaptureResponseRouterData<PaymentCaptureResponse>>
         } else {
             (enums::AttemptStatus::Pending, None)
         };
+
+        let resource_id = if item.data.request.multiple_capture_data.is_some() {
```
Comment (ArjunKarthik) [inline crates/router/src/connector/checkout/transformers.rs:626]: use item.request.is_multiple_capture()

Comment (ArjunKarthik) [inline crates/router/src/connector/checkout/transformers.rs:626]: Add comments on why we need this if else statement

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

**Review by ArjunKarthik — DISMISSED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Pull request merged**